MusicalModel.md

```diff
@@ -0,0 +1,183 @@
+# Musical Model
+
+## Elements
+
+The most basic musical attributes, from `Pitch` to `Articulation` to `DynamicMarking` to `OSCMessage` values.
+
+Some abbreviated examples:
+
+```Swift
+struct Pitch {
+    let value: Float
+}
+
+struct Articulation: OptionSetType {
+    let accent: ...
+    let staccato: ...
+    let tenuto: ...
+    ...
+}
+
+struct DynamicMarking {
+
+    enum Element {
+        case p,f,o,m,s,z
+    }
+
+    let elements: [Element]
+}
+
+struct OSCMessage {
+    let addressPattern: String
+    let arguments: [String]
+}
+```
+
+These can be contextualized with the performer and instrument to which these elements apply:
+
+```Swift
+struct Event {
+    let performerIdentifier: String
+    let instrumentIdentifier: String
+    let elements: [Any] // defined above
+}
+```
+
+We can leave `Event.elements` declared as an `Array` of `Any` values. In that case, anything can be passed in here, then we will just destructure the type as is necessary on the backend.
+
+## Rhythm Model
+
+The elements above will need to be contextualized within some temporal model. Commonly in western notated music, we use a hierarchical division of time often called _rhythm_.
+
+### Rhythm in abstract
+
+We can call the most basic element of _rhythm_ a `MetricalDuration`, with an abbreviated example:
+
+```Swift
+struct MetricalDuration {
+    let beats: Int
+    let subdivision: Int
+}
+```
+
+To put this in this in a richer abstract representation, we need to extend certain rhythms to make compound rhythmical events. We can either provide a "tie" to the previous event, or provide a new _something_.
+
+```Swift
+enum TieOrNote <T> {
+    case tie
+    case note(T)
+}
+```
+
+### Placing events in durational context
+
+In the most abstract contexts, we don't need to know if there are rests or proper events. In the more practical example of notated music, we can either do something at given metrical duration offset, or not:
+
+```Swift
+enum RestOrEvent <T> {
+    case rest
+    case event(T)
+}
+```
+
+> Both `RestOrEvent<T>` and `TieOrNote<T>` are essentially the `Optional` monad. 
+>
+> We can discuss the desirability of using more domain-specific definitions of these types or the (initial) elegance of `T?`.
```

Comment:
If we just throw around `T?` (`Optional<T>`) values, it then creates the circumstances of `??` (doubly-optional) values. This feels like it is getting out of hand.